src/RuleSet/RuleSet.php

```diff
@@ -104,9 +104,22 @@ public function addRule(Rule $ruleToAdd, ?Rule $sibling = null): void
         $position = \count($this->rules[$propertyName]);
 
         if ($sibling !== null) {
+            $siblingIsInSet = false;
             $siblingPosition = \array_search($sibling, $this->rules[$propertyName], true);
             if ($siblingPosition !== false) {
+                $siblingIsInSet = true;
                 $position = $siblingPosition;
+            } elseif ($siblingIsInSet = $this->hasRule($sibling)) {
```

Comment:
Please let's separate the assignment and the check into two statements so that the condition is side-effect free.

Reply:
Done.